src/Pkg3.jl

```diff
@@ -8,12 +8,19 @@ else
     import Dates
 end
 
-const DEPOTS = [joinpath(homedir(), ".julia")]
-depots() = DEPOTS
-logdir() = joinpath(DEPOTS[1], "logs")
+@enum LoadErrorChoice LOAD_ERROR_QUERY LOAD_ERROR_INSTALL LOAD_ERROR_ERROR
 
-const USE_LIBGIT2_FOR_ALL_DOWNLOADS = false
-const NUM_CONCURRENT_DOWNLOADS      = 8
+Base.@kwdef mutable struct GlobalSettings
```

Comment:
I didn't know `Base.@kwdef`, that's awesome! you should have told me I can use it for `Base.REPL.Options` :) Speaking of which, would `Base.REPL.Settings` be a better name for it?
And here... wouldn't `Settings` be a better name than `GlobalSettings`? for the `const` global object, `GLOBAL_SETTINGS` looks like a good name, but could we imagine people doing crazy things and using multiple distinct instances of `GlobalSettings` concurrently, making it more of a local thing? (I really don't know Pkg3's code, so I speak blindly here).

Reply:
Yeah, perhaps `Settings` is a better name. I don't really think many instances of it will be used though.